### DISCUSSION TITLE
next.js app taking 15-25 seconds to load from “cold” on google app engine

### DISCUSSION THREAD

**Discussion by alexfoxy** [vercel/next.js#12003] [Help]
I am deploying a next.js app to a Google App Engine (standard) but I am having trouble with cold start times. 

I am building locally then deploying the `.next` folder to GAE. On the first boot it takes 15-25 seconds to boot up. Looking at the logs I can see a log for a very long `GET 200` request but there's not really any information about what is going on. See log below:

<img width="1389" alt="Screenshot 2020-04-18 at 17 13 29" src="https://user-images.githubusercontent.com/151772/79643403-b895e780-819a-11ea-9e7f-24851948df4a.png">

When I run `yarn start` locally it boots up in < 1, which as far as I understand is running the same production build? It is almost as though GAE is doing a build before it eventually deploys..

Any pointers would be immensely appreciated!

Alex

**Comment by timneutkens** [comment]
Sounds like something related to how Google App Engine works, for example it could have a cold-boot problem where it has to boot up the container on the first request.

**Reply by alexfoxy** [reply]
That appears to be something to do with it. I can mitigate the issue by keeping the server alive, but I wondered if anyone else has seen this issue.

I would have thought all it needed to do was run ‘next start’ which takes less than a second on my local machine.

**Reply by timneutkens** [reply]
Yeah so generally these platforms need to boot up the container too which takes time. You might have a look at https://zeit.co which turns every page that does SSR into a serverless function.

**Reply by alexfoxy** [reply]
Yep, I've actually enabled serverless deployment and I get much quicker response times now. I didn't realise that when you deployed on now it enabled this by defualt.

**Reply by mashaole** [reply]
How did yo enable that in Nextjs?

**Reply by alexfoxy** [reply]
@mashaole you have to enable it in the AppEngine config. https://cloud.google.com/appengine/docs/standard/nodejs/configuring-warmup-requests

**Reply by mashaole** [reply]
> @mashaole you have to enable it in the AppEngine config. https://cloud.google.com/appengine/docs/standard/nodejs/configuring-warmup-requests

All you need to add is
 inbound_services:
- warmup 
in the .yaml right ?
